src/App.tsx

```diff
@@ -15,37 +15,90 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+const App: React.FC = () => {
+  const [list, setList] = useState<string[]>(goodsFromServer);
+  const [reverseList, setReverseList] = useState<boolean>(false);
+  const [sortedState, setSortedState] = useState<number>(0);
+
+  const reset = () => {
+    setList(goodsFromServer);
+    setReverseList(false);
+    setSortedState(0);
+  };
+
+  const sortedGoods = (goods: string[]) => {
+    setSortedState(1);
+    const result = [...goods].sort((a, b) => a.localeCompare(b));
+
+    if (reverseList) {
+      return result.reverse();
+    }
+
+    return result;
+  };
+
+  const sortedGoodsLength = (goods: string[]) => {
```

Comment:
The state variables are available for every function inside a component.
In this case, you don't need to create every sorting function receiving "goods" as parameter, because everytime you call it, you are passing the value of "goods" as the state value for "list".
You can simply remove the parameters from your sorting functions and let them access the state direcly